File: binance4j-core/src/main/java/com/binance4j/core/order/SideEffectType.java
package com.binance4j.core.order;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Desired side-effect for margin orders
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public enum SideEffectType {
    /** normal trade order */
    NO_SIDE_EFFECT,
    /** margin trade order */
    MARGIN_BUY,
    /** for making auto repayment after order filled */
    AUTO_REPAY
}

File: connectors/src/main/kotlin/com/binance4j/margin/dto/SideEffectType.kt
package com.binance4j.margin.dto;

/** Desired side-effect for margin orders */
public enum SideEffectType {
	/** normal trade order. */
	NO_SIDE_EFFECT,
	/** margin trade order. */
	MARGIN_BUY,
	/** for making auto repayment after order filled. */
	AUTO_REPAY
}
